wrapper/src/main/java/software/amazon/jdbc/exceptions/MultiAzDbClusterPgExceptionHandler.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.exceptions;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class MultiAzDbClusterPgExceptionHandler implements ExceptionHandler {
+
+  public static final List<String> NETWORK_ERRORS = Arrays.asList(
+      "28000", // 28000 for access denied during reboot, this should be considered as a temporary failure
+      "53", // insufficient resources
+      "57P01", // admin shutdown
+      "57P02", // crash shutdown
+      "57P03", // cannot connect now
+      "58", // system error (backend)
+      "08", // connection error
+      "99", // unexpected error
+      "F0", // configuration file error (backend)
+      "XX" // internal error (backend)
+  );
+
+  public static final List<String> ACCESS_ERROR = Collections.singletonList(
+      "28P01"
+  );
+
+  @Override
+  public boolean isNetworkException(final Throwable throwable) {
+    Throwable exception = throwable;
+
+    while (exception != null) {
```

Comment:
Does that mean that we inspect all the layers of errors until either we reach the end, either get a SQLException?

Reply:
> Does that mean that we inspect all the layers of errors until either we reach the end, either get a SQLException?
Yes. This is align with the default PgExceptionhandler.

> how do we handle this in other dialects ?
The only difference between MultiAZ handler and default Pg handler is, they take the error code `28000` differently.

* In other clusters (like Aurora), the `28000` means "access denied by IAM auth", leading to a fatal error.
* But in MultiAZ DB Cluster, The PG may raise `28000` during cluster rebooting, meaning "deny access and queries due to rebooting". The driver should retry and keep on the failover logic.

As a result, the IAM auth plugin will be incompatitable with MultiAZ db cluster. Because the driver can not distinguish the scenario of `28000` in a decent way in the current design.

Afterall, a refactor is expected because the existing class `PgExceptionHandler` and `MultiAzDbClusterPgExceptionHandler` are essentially the same. I would keep the refactor out of currrent PR for easier review.